[llvm][ARM][NFC] Renaming FeaturePrefLoopLogAlign

nasherm opened the pull request:
The feature 'FeaturePrefLoopLogAlign' was misleading as it was used to set the alignment of branch instructions such as functions. Renamed to FeaturePrefBranchInstAlignment.

nasherm committed:
[llvm][ARM][NFC] Renaming FeaturePrefLoopLogAlign

The feature 'FeaturePrefLoopLogAlign' was misleading
as it was used to set the alignment of branch instructions
such as functions. Renamed to FeaturePrefBranchInstAlignment.

Change-Id: I88edcf2fe946d50fc33765273d36e1abcbe9d959

davemgreen commented on llvm/lib/Target/ARM/ARMFeatures.td:
```diff
@@ -372,11 +372,11 @@ def FeatureVMLxForwarding : SubtargetFeature<"vmlx-forwarding",
-def FeaturePrefLoopAlign32 : SubtargetFeature<"loop-align", "PrefLoopLogAlignment","2",
-                                              "Prefer 32-bit alignment for loops">;
+def FeaturePrefBranchInstAlign32 : SubtargetFeature<"loop-align", "PrefBranchInstLogAlignment","2",
+                                              "Prefer 32-bit alignment for branch instructions (loops, functions)">;
```
branch instructions -> branch targets

nasherm replied:
Done

davemgreen reviewed (approved):
Thanks for doing this. I think I would go with "BranchAlign" instead of "BranchInstAlign", which implies that the actual br is aligned.

Otherwise LGTM.

nasherm committed:
Review comments

Change-Id: If0549d2bd62209279f5d21b90609ad42a1d116d4

davemgreen reviewed (approved):
Thanks

tommat01 reviewed (commented):
Personally I would just kept it consistent with the subtarget feature spelling, e.g. `FeatureLoopAlign` to match `loop-align`. But I don't feel that strongly. If you're going to stick with the current suggestion then maybe expand "Pref" to "Prefer" since it only saves two characters.

nasherm committed:
Clang formatting

Change-Id: Ibf1ffab27184a12a5d0fdab2ec7af437192552b5

nasherm commented:
> Personally I would just kept it consistent with the subtarget feature spelling, e.g. FeatureLoopAlign to match loop-align. But I don't feel that strongly. If you're going to stick with the current suggestion then maybe expand "Pref" to "Prefer" since it only saves two characters.

I've renamed `loop-align` to `branch-align` to keep it consistent. Also changed "Pref" to "Prefer"

tmatheson-arm commented:
> I've renamed `loop-align` to `branch-align` to keep it consistent. Also changed "Pref" to "Prefer"

Unfortunately you can't do that; it will break any existing IR that refers to this subtarget feature as `loop-align`.

nasherm commented:
> > I've renamed `loop-align` to `branch-align` to keep it consistent.
> 
> Unfortunately you can't do that; it will break any existing IR that refers to this subtarget feature as `loop-align`.

Ah I see, thanks for pointing this out.

nasherm committed:
Typo. Can't rename 'loop-align'
